frontend/src/pages/details-page.tsx

```diff
@@ -1,14 +1,35 @@
-import { useLocation, useNavigate } from 'react-router-dom';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
 import navigateBackWhiteIcon from '@/assets/svg/navigate-back-white.svg';
-import Post from '@/types/post-type';
+// import Post from '@/types/post-type';
 import formatPostTime from '@/utils/format-post-time';
 import CategoryPill from '@/components/category-pill';
+import {useEffect, useState} from "react";
+import axios from "axios";
 
 export default function DetailsPage() {
   const { state } = useLocation();
-  const post: Post = state?.post;
+  const [post,setPost] = useState(state?.post);
+  const initialVal = post === undefined;
+  const [loading, setIsLoading] = useState(initialVal);
+  const {postId} = useParams();
   const navigate = useNavigate();
-  return (
+
+  useEffect(()=>{
+    const getPostById = async() =>{
+      try{
+        await axios.get(import.meta.env.VITE_API_PATH + `/api/posts/${postId}`).then((response)=>{
+          console.log(response.data);
+          setIsLoading(false);
+          setPost(response.data);
+        })
+      }catch (error){
+        console.log(error);
+      }
+    }
+    if(post===undefined){getPostById();}
+  },[post])
+
+  if(!loading)return (
     <div className="min-h-screen bg-light dark:bg-dark">
```

Comment:
Kindly run the npm run format on the frontend, so that the code is readable and formatted nicely.